### PR TITLE
perf(server): add FlushConsolidationHandler

### PR DESCRIPTION
**Pull request by amitksingh1490**

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/service/server/ServerChannelInitializer.scala [hunk 1]
```diff
@@ -38,6 +39,10 @@ final case class ServerChannelInitializer[R](
     // Add Keep-Alive handler is settings is true
     if (cfg.keepAlive) pipeline.addLast(HTTP_KEEPALIVE_HANDLER, new HttpServerKeepAliveHandler)
 
+    // FlushConsolidationHandler
+    // Always add FlushConsolidationHandler, optimises the flush calls
+    pipeline.addLast(HTTP_SERVER_FLUSH_CONSOLIDATION, new FlushConsolidationHandler)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/server/ServerChannelInitializer.scala]: keep it as a config.

File: zio-http/src/main/scala/zhttp/service/Server.scala [hunk 2]
```diff
@@ -76,6 +79,7 @@ object Server {
   val advancedLeakDetection: UServer = LeakDetection(LeakDetectionLevel.ADVANCED)
   val paranoidLeakDetection: UServer = LeakDetection(LeakDetectionLevel.PARANOID)
   val keepAlive: UServer             = KeepAlive
+  val flushConsolidation: UServer    = FlushConsolidation
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Server.scala]:
```suggestion
  val consolidateFlush: UServer    = FlushConsolidation
```

File: zio-http/src/main/scala/zhttp/service/Server.scala [hunk 3]
```diff
@@ -26,6 +26,7 @@ sealed trait Server[-R, +E] { self =>
     case Address(address)     => s.copy(address = address)
     case AcceptContinue       => s.copy(acceptContinue = true)
     case KeepAlive            => s.copy(keepAlive = true)
+    case FlushConsolidation   => s.copy(flushConsolidation = true)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Server.scala]:
```suggestion
    case ConsolidateFlush   => s.copy(consolidateFlush = true)
```

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 437421.50**

**Comment by tusharmath** [comment]
I want to see the impact on performance when we use this setting.

**Comment by tusharmath** [comment]
I think it will be easier if we get #615 integrated for this.

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 428838.81**

File: zio-http/src/main/scala/zhttp/service/server/ServerChannelInitializer.scala [hunk 4]
```diff
@@ -52,6 +53,10 @@ final case class ServerChannelInitializer[R](
     // For reference: https://netty.io/4.1/api/io/netty/handler/flow/FlowControlHandler.html
     if (cfg.flowControl) pipeline.addLast(FLOW_CONTROL_HANDLER, new FlowControlHandler())
 
+    // FlushConsolidationHandler
+    // Add FlushConsolidationHandler, optimises the flush calls
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/server/ServerChannelInitializer.scala]:
```suggestion
    // Flushing content is done in batches. Can potentially improve performance.
```

**Review by tusharmath — APPROVED** [review]

**Pull request merged**